File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for i in tokens:
            if i == "+":
                pop1 = int(stack.pop())
                pop2 = int(stack.pop())
                stack.append(pop1 + pop2)
            elif i == "-":
                pop1 = int(stack.pop())
                pop2 = int(stack.pop())
                stack.append(pop2 - pop1)
            elif i == "*":
                pop1 = int(stack.pop())
                pop2 = int(stack.pop())
                stack.append(pop1 * pop2)
            elif i == "/":
                pop1 = int(stack.pop())
                pop2 = int(stack.pop())
                stack.append(pop2 / pop1)
            else:
                stack.append(i)
        return int(stack.pop())

File: Data Structures & Algorithms/validate-parentheses/submission-2.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> brackets = new Stack<>();
        for(int i = 0; i < s.length(); i++){
            char c = s.charAt(i);
            if(c == '[' || c== '(' || c == '{'){
                brackets.push(c);
            }
            else{
                if(brackets.isEmpty()){
                    return false;
                }
                char popped = brackets.pop();
                if((c == ')' && popped != '(') || 
                (c == ']' && popped != '[') || 
                (c == '}' && popped != '{')){
                    return false;
                }
            }
        }
        return brackets.isEmpty();
    }
}
